### ISSUE TITLE
app/vmselect：bottomk doesn't return any series if there are two half series in query time range but topk does

### ISSUE THREAD

**Issue by xiaozongyang** [VictoriaMetrics/VictoriaMetrics#5506]
### Describe the bug

We have two group of series shown as following picture, one group disappeared after some time and the other appeared and we have common labels on those groups. It's caused by a deployment operation.

#### Orignial series
![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/8848479/42626c2d-12a3-4d56-ae86-88e8c229f0bb)
There are five original series without any aggregation function.

#### Topk Result
![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/8848479/dd9cf8d1-bc9e-4e1a-83f3-9646179513d2)
The `topk` function return 1 series which matches the expection.

#### Botoom Result
![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/8848479/b93e7b92-419a-48bc-bf21-387a233d37fb)
And the information of grafana query inspector is
```
{
  "request": {
    "url": "api/datasources/proxy/175/api/v1/query",
    "method": "POST",
    "data": {
      "query": "bottomk(1, flink_taskmanager_job_task_operator_numRecordsInPerSecond{job_name=~\"octopus_engine_write_proxy.*\", job_name=\"octopus_engine_write_proxy:N7FfAFpB6X1W_trace_router_ali\", subtask_index=\"1\"})",
      "time": 1703082016
    },
    "hideFromInspector": false
  },
  "response": {
    "status": "success",
    "isPartial": false,
    "data": {
      "resultType": "vector",
      "result": []
    }
  }
}
```

#### Expected Behavior
The `bottomk` function return 0 series which DOESN'T matches exepection. Becuase we have five series here, we want some series if there are original series. The expected result maybe following:

![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/8848479/37164cd2-20a3-412c-ab93-67027d417f3b)

### To Reproduce
1. insert following metrics into vm with `POST /api/v1/import`
```
{"metric":{"__name__":"foo","job":"job1","service": "test", "instance":"localhost:8080"},"values":[1, 2, 3],"timestamps":[1703174400000, 1703174415000, 1703174430000]}
{"metric":{"__name__":"foo","job":"job1","service": "test", "instance":"localhost:8081"},"values":[2, 3, 4],"timestamps":[1703174400000, 1703174415000, 1703174430000]}
{"metric":{"__name__":"foo","job":"job1","service": "test", "instance":"localhost:8082"},"values":[3, 4, 5],"timestamps":[1703174400000, 1703174415000, 1703174430000]}
{"metric":{"__name__":"foo","job":"job1","service": "test", "instance":"localhost:8083"},"values":[3, 2, 1],"timestamps":[1703174550000, 1703174565000, 1703174580000]}
{"metric":{"__name__":"foo","job":"job1","service": "test", "instance":"localhost:8084"},"values":[4, 3, 2],"timestamps":[1703174550000, 1703174565000, 1703174580000]}
{"metric":{"__name__":"foo","job":"job1","service": "test", "instance":"localhost:8085"},"values":[5, 4, 3],"timestamps":[1703174550000, 1703174565000, 1703174580000]}
```
2. open vmui on `http://localhost:8428`
3. try following queries in above time range
```
foo{service="test"}
topk(2, foo{service="test"})
bottomk(2, foo{service="test"})
```

### Version

vmselect-20230620-062146-v1.87.6-cluster-0-g468cd5f

### Additional information

I check source code, it seems like that is related about the function `lessWithNaNs` in file [app/vmselect/promql/aggr.go](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/app/vmselect/promql/aggr.go#L1255). The functions sort values like `NaN < a < b`.

```
func lessWithNaNs(a, b float64) bool {
	if math.IsNaN(a) {
		return !math.IsNaN(b)
	}
	return a < b
}
```

**Comment by xiaozongyang** [comment]
ping @valyala

**Comment by valyala** [comment]
FYI, the fix for this issue has been included in [v1.87.13 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.87.13). The fix will be included in the latest release too.

**Comment by valyala** [comment]
FYI the fix for this issue has been included in [v1.93.10 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.93.10).

**Comment by xiaozongyang** [comment]
@valyala Thanks your works, and this issue will be closed.

**Comment by valyala** [comment]
FYI, the fix for the issue has been included in [v1.97.0](https://docs.victoriametrics.com/CHANGELOG.html).
